Aplib.Tests/Desire/GoalTests.cs

```diff
@@ -0,0 +1,84 @@
+using Aplib.Core.Desire;
+using Aplib.Tests.Stubs.Desire;
+using FluentAssertions;
+
+namespace Aplib.Tests.Desire;
+
+public class GoalTests
+{
+    [Fact]
+    public void Goal_WhenConstructed_ContainsCorrectMetaData()
+    {
+        // Arrange
+        Tactic tactic = new TacticStub(() => { });
+        IGoalPredicate predicate = new ConstantGoalPredicate(0f);
+        var name = "Such a good goal name";
+        var description = "\"A lie is just a good story that someone ruined with the truth.\" - Barney Stinson";
```

Comment:
Nitpick: I would argue `var` instead of `string` does not improve code readability. Also see [C# at Google Style Guide: the var keyword](https://google.github.io/styleguide/csharp-style.html#the-var-keyword).

Reply:
Good call